### PR TITLE
[US-4470] twig cache - improve performance

### PR DESCRIPTION
**Pull request by simara-svatopluk**
| Q             | A
| ------------- | ---
|Description, reason for the PR| US-4470 - we want to improve the performance of the shop easily
|New feature| Yes
|BC breaks| No
|Fixes issues| 
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

**Review by TomasLudvik — APPROVED** [review]
Hello @simara-svatopluk,

good job there. I have found no problems.

File: src/Shopsys/ShopBundle/Resources/views/Front/Layout/layoutWithPanel.html.twig [hunk 1]
```diff
@@ -13,7 +13,9 @@
         </div>
 
         <div class="web__main__panel">
-            {{ render(controller('ShopsysShopBundle:Front/Category:panel', { request: app.request } )) }}
+            {% cache 'category_panel'~app.request.pathinfo 5*60 %}
```
Comment (PetrHeinz) [inline src/Shopsys/ShopBundle/Resources/views/Front/Layout/layoutWithPanel.html.twig:16]: So this is cached individually for every single page? That seems a bit wasteful to me.

Comment (simara-svatopluk) [reply src/Shopsys/ShopBundle/Resources/views/Front/Layout/layoutWithPanel.html.twig:16]: Yes it is, but in different categories the panel looks different. This is a quick solution

**Comment by PetrHeinz** [comment]
I've tested the performance impact and it looks fine, improving the page load by around 15 % (having more impact when more users are browsing the web at the same time).

image

It leads to having links to 404 page for up to 5 minutes when, for example, a category is deleted, but this is something we knew beforehand.
A large amount of cached fragment might be created when there is a lot of pages, but with only demo data there is not enough pages for this to be a problem.

The keys in redis look a bit weird (eg. `twig_[TwigCache__category_panel/electronics/__onDomain1][1]`) but it has no impact on the functionality...

File: src/Shopsys/ShopBundle/Resources/config/services.yml [hunk 2]
```diff
@@ -42,3 +42,21 @@ services:
...
+    class: Shopsys\ShopBundle\Twig\CurrentDomainLifetimeCacheStrategy
+    arguments:
+      - '@twig_cache_extension.cache_provider'
+
+  twig_cache_extension:
+    class: Asm89\Twig\CacheExtension\Extension
+    arguments:
+      - '@twig_cache_extension.cache_strategy'
+    tags:
+      - { name: twig.extension }
```
Comment (TomasVotruba) [inline src/Shopsys/ShopBundle/Resources/config/services.yml:62]: These tags are already covered by autoconfiguration: https://github.com/shopsys/demoshop/pull/9/files#diff-4314c4cdb737e9a7ed5c1dc455dabaafR7

It makes no sense to use them duplicitly

Comment (simara-svatopluk) [reply src/Shopsys/ShopBundle/Resources/config/services.yml:62]: Well it may have sense, could You please create an issue? ;-)

Comment (TomasVotruba) [reply src/Shopsys/ShopBundle/Resources/config/services.yml:62]: Nono, it's just my code-review comment. You can handle it if it seems important

**Pull request merged**